### DISCUSSION TITLE
[Thesis] AI-Monster

### DISCUSSION THREAD

**Discussion by cagix** [Programmiermethoden/Dungeon#1372] [Ideas]
Monster mit AI: Zusammenarbeit, Lernfähigkeit => Beispiele, Systemdesign, Interfaces

**Comment by AMatutat** [comment]
Hier muss auch mal das Pathfinding angepasst werden.
Bisher war das immer tile-based, aber mittlerweile reicht das nicht mehr aus.
Tiles können begehbar sein, aber von Objekten blockiert werden (z. B. ein Schrank, der im Weg steht).
Auch verändert sich das Level, das heißt der "Graph", der für das Pathfinding verwendet wird, muss dynamisch angepasst werden und nicht nur einmal zu Beginn des Levels.

Dann gibt es Tiles, die nur von bestimmten Monstern begehbar sein sollen (fliegende Gegner können über Pits fliegen).
Evtl. möchte ich Monstern auch bestimmte Tiles verbieten (z. B. soll nicht jedes Monster durch eine Tür gehen können).
Oder evtl. will ich Geister, die durch Mauern gehen können.

Und dann kann man grundsätzlich mal mehr Monster mit spezifischem Verhalten bauen.
Ganz spontan:

* Doc, der andere Monster heilt
* fliegende Gegner
* Eisgolem, der nur mit Feuer besiegt werden kann
* Feuergolem, der mit Feuer geheilt wird

**Comment by cagix** [comment]
brauchen wir weder für Produs noch für Escape.
